### PR TITLE
Update README.md for use with Deno 1.4.0

### PR DESCRIPTION
**Pull request by ianfabs**
# TL;DR
  - Added example
  - Updated README.md to work with Deno v1.4.0

I haven't done extensive testing with this yet, but firstly *thank you so much* for creating this. Saved me an hour of copying and pasting.

I was following the readme as it were, and `deno-dom-wasm.ts` failed repeatedly with a couple of errors. The first error I got was something about `importsNotUsedAsValues` being set to `"error"`:

image

Which I fixed by creating a `tsconfig.json` that looks like this:
```json
{
  "compilerOptions": {
    "importsNotUsedAsValues": "remove"
  }
}
```
After that was fixed, then I got this type error about `NodeType`:

image

Which I fixed by switching to `deno-dom-native`. But then I got an error about the `--isolatedModules` flag, which I solved by adding the following to the `tsconfig.json`:
## `tsconfig.json` changes
```json
{
  "compilerOptions": {
    "importsNotUsedAsValues": "remove",
    "isolatedModules": false
  }
}
```
## Error
image

I wasn't out of the woods yet though, as my good old friend `NodeType` error resurfaced. 
image

I changed to the latest version - `v0.1.2-alpha4` - and presto, it worked!

Though it has this fun side-effect, that it creates a bunch of empty space in the terminal. But that didn't impact it from working! Thank you again for creating this! Keep up the good work!!!!

## The command used to run the script:
```sh
deno run -c ./tsconfig.json -A --unstable ./mod.ts
```
**Explaination**:
  - The `-c ./tsconfig.json` tells deno to run this script with the specified typescript compiler config
  - `-A`, short for `--allow-all`, gives the deno script to be run ***all permissions***, including: `--allow-env`, `--allow-net`, `--allow-plugin`, and in my case I also needed `--allow-read`, `--allow-write`.
  - `--unstable` enables the parts of the unstable deno API that are required for use.
  - And finally, `./mod.ts` would be the path to your deno script.

### CODE REVIEW

File: examples/basic/mod.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { DOMParser, Element } from "https://deno.land/x/deno_dom@v0.1.2-alpha4/deno-dom-native.ts";
```
Comment (b-fuze) [inline examples/basic/mod.ts:1]: The examples are a nice idea, just that I'd import from `https://deno.land/x/deno_dom/deno-dom-wasm.ts` here.

File: examples/basic/run [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+deno run -c ./tsconfig.json -A --unstable ./mod.ts
```
Comment (b-fuze) [inline examples/basic/run:2]: Deno is cross-platform so I'd rather avoid including shell scripts that may seem required to run a Deno DOM-dependent script

Comment (CodingCarter) [inline examples/basic/run:2]: I, personally, think that we shouldn't be using `-A` or `--allow-all` instead of just adding `--allow-plugin --unstable` because we don't need all of those extra permissions and it adds insecurity as other modules would then be able to access all of that information.

Comment (CodingCarter) [reply examples/basic/run:2]: Also, there is another instance on line 38 of the README.

File: README.md [hunk 3]
```diff
@@ -4,8 +4,10 @@ An implementation of the browser DOM—primarily for SSR—in Deno. Implemented
 Rust, WASM, and obviously, Deno/TypeScript.
 
 ## Example
+
+*mod.ts*
```
Comment (b-fuze) [inline README.md:8]: I don't think this is necessary

**Review by CodingCarter — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -4,8 +4,10 @@ An implementation of the browser DOM—primarily for SSR—in Deno. Implemented
...
 ```typescript
-import { DOMParser, Element } from "https://deno.land/x/deno_dom@v0.1.2-alpha3/deno-dom-wasm.ts";
+import { DOMParser, Element } from "https://deno.land/x/deno_dom@v0.1.2-alpha4/deno-dom-native.ts";
```
Comment (b-fuze) [inline README.md:10]: Please import from `https://deno.land/x/deno_dom/deno-dom-wasm.ts`. I primarily advertise the WASM backend because it's cross-platform and needs no permissions to run. The native backend is primarily for its performance advantages, but because of all the Deno permissions it needs, it shouldn't be the preferred backend.

File: README.md [hunk 5]
```diff
@@ -5,7 +5,7 @@ Rust, WASM, and obviously, Deno/TypeScript.
 
 ## Example
 ```typescript
-import { DOMParser, Element } from "https://deno.land/x/deno_dom@v0.1.2-alpha3/deno-dom-wasm.ts";
+import { DOMParser, Element } from "https://deno.land/x/deno_dom@v0.1.2-alpha4/deno-dom-native.ts";
```
[the remark this answers is not in the document]
Comment (CodingCarter) [reply README.md:10]: This may not be required because of the .tsconfig, though.

File: README.md [hunk 6]
```diff
@@ -21,6 +23,28 @@ p.innerHTML = "DOM in <b>Deno</b> is pretty cool";
...
+deno run -c ./tsconfig.json -A --unstable ./mod.ts
+```
+
+You should see output like:
+```
+Hello from Deno!
+Deno!
+<html>DOM in <b>Deno</b> is pretty cool</html>
+```
+
```
Comment (b-fuze) [inline README.md:47]: This isn't a bad idea, but the `tsconfig.json` isn't required here, and the command should be
```sh
deno run ./examples/basic/mod.ts
```
to align more with your example.

**Review by b-fuze — COMMENTED** [review]
Please review these comments

File: README.md (hunk 4 above)
Comment (b-fuze) [reply README.md:10]: And yeah, that should also be
```js
import { DOMParser,  } from "https://deno.land/x/deno_dom/deno-dom-wasm.ts";
import type { Element } from "https://deno.land/x/deno_dom/deno-dom-wasm.ts";
```
to be prepared for Deno 1.5.0

File: examples/basic/mod.ts (hunk 1 above)
Comment (b-fuze) [reply examples/basic/mod.ts:1]: This should also be
```typescript
import { DOMParser } from "https://deno.land/x/deno_dom/deno-dom-wasm.ts";
import type { Element } from "https://deno.land/x/deno_dom/deno-dom-wasm.ts";
```

File: examples/basic/run (hunk 2 above)
Comment (ianfabs) [reply examples/basic/run:2]: You're right! I should have been more explicit.

File: README.md (hunk 4 above)
Comment (ianfabs) [reply README.md:10]: You're absolutely right. I've almost finished fixing all the nits, however to get examples working properly a **tsconfig.json** with `importsNotUsedAsValues` set to `remove` is required.

File: README.md (hunk 6 above)
Comment (ianfabs) [reply README.md:47]: Using `deno run ./examples/basic/mod.ts` would not successfully launch the example in this case :(

Comment (ianfabs) [reply README.md:47]: I may stand corrected, hold on

**Comment by b-fuze** [comment]
Btw, you need to merge my recent changes to see why WASM isn't required here.

File: README.md (hunk 6 above)
Comment (ianfabs) [reply README.md:37]: I did indeed, this should look good now

File: examples/basic/mod.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { DOMParser } from "https://deno.land/x/deno_dom@v0.1.2-alpha4/deno-dom-wasm.ts";
+import type { Element } from "https://deno.land/x/deno_dom@v0.1.2-alpha4/deno-dom-wasm.ts";
```
Comment (b-fuze) [inline examples/basic/mod.ts:2]: Please remove the specific version `@v0.1.2` from the imports in favor of implicit version selection

File: examples/basic/run (hunk 2 above)
Comment (0kku) [reply examples/basic/run]: FYI: Allowing native plugins is just as insecure as `-A` — native plugins are not sandboxed, so it's the same as having all permissions for them. I still think being explicit is good though. Makes it easier for people to understand what's going on. Don't let yourself be cradled in a false sense of security: allowing plugins means all bets are off, and that's what the WASM version exists for.

File: examples/basic/mod.ts (hunk 7 above)
Comment (b-fuze) [reply examples/basic/mod.ts:2]: Also, rename this file to `examples/basic.ts`  we don't need a folder here

File: examples/basic/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+  "compilerOptions": {
+    "importsNotUsedAsValues": "remove",
+  }
+}
```
Comment (b-fuze) [inline examples/basic/tsconfig.json:5]: We don't need this anymore

**Comment by Hunam6** [comment]
@ianfabs maybe you forgot about this PR
